### ISSUE TITLE
Resource status not updating correctly

### ISSUE THREAD

**Issue by bourgeoisor** [pravega/zookeeper-operator#36]
Although there is a concept of resource status for the ZookeeperCluster, the status is never updated throughout the lifecycle of the reconciliation.
We should be updating the ZookeeperCluster status with the correct node count of the stateful set here: https://github.com/pravega/zookeeper-operator/blob/master/pkg/zk/sync.go#L38-L44

This is the current status of a fully deployed ZookeeperCluster:
```
~ kubectl get zookeeper-cluster some-zookeeper-cluster -n some-namespace -o json
. . .
    "status": {
        "members": {
            "ready": null,
            "unready": null
        },
        "size": 0
    }
. . .
```

**Comment by finiks** [comment]
@adrianmo what's the story behind the `members`' ready and unready arrays of string? Is that still needed, and if so, what strings should populate it?
I personally would've just gone with a simple `replicasAvailable` count which could be then cross-referenced with the Spec's `replicas` (or `size`) count to know if the cluster is fully available or not. That said, there might be reasoning behind going more in depth for this resource.

**Comment by finiks** [comment]
#37 got merged in, which implements a very simple iteration of cluster status.
We'll need to go a bit more in depth in the future. Leaving this issue open.

**Comment by spiegela** [comment]
Resolved per #41
